gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -168,13 +168,16 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
         #}
         # TODO(yon-mg): handle nested fields corerctly rather than using only top level fields
         #               not required for GCE
-        potentialParams = {
+        query_params = {
             {%- for field in method.query_params %}
             '{{ field|camel_case }}': request.{{ field }},
             {%- endfor %}
         }
-        potentialParams = ((k, v) for k, v in potentialParams.items() if v)
-        for i, (param_name, param_value) in enumerate(potentialParams):
+        # TODO(yon-mg): further discussion needed whether 'python truthiness' is appropriate here
+        #               discards default values
+        # TODO(yon-mg): add test for proper url encoded strings
+        query_params = ((k, v) for k, v in query_params.items() if v)
+        for i, (param_name, param_value) in enumerate(query_params):
```

Comment:
nit: Do consider the string.Join suggestion from earlier (but it's certainly not a blocker)